scripts/excel_utils.py

```
@@ -0,0 +1,192 @@
+
+    return str(output_file)
+
+
+def csv_to_xlsx(input_path: str, output_dir: str, delimiter: str = ",") -> str:
+    """Convert a CSV file into a single-sheet .xlsx workbook."""
+    input_file = Path(input_path)
+    output_file = Path(output_dir) / f"{input_file.stem}.xlsx"
+
+    delimiter = delimiter or ","
+    if len(delimiter) > 2:
+        raise ValueError("delimiter must be a single character (use ',' '\\t' ';' '|').")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)